code/modules/mob/living/death.dm

```diff
@@ -74,6 +74,8 @@
 	timeofdeath = world.time
 	tod = station_time_timestamp()
 	var/turf/T = get_turf(src)
+	if(SSmapping.level_trait(T.z, ZTRAIT_SECRET) && !client?.holder)
```

Comment:
This is bad way to stop players from peeking behind the walls while dead. It'll effectively impair revival of the mob considering it requires the player to be in control of the mob and not a ghost.
That said, it's better to just do some changes to `update_sight()` so it won't grant (SEE_MOBS|SEE_OBJS|SEE_TURFS) to dead mobs while they're in a sekrit zlevel.

Reply:
not true defibs no longer require ghost to be in body and you can return to it when your body comes back from death anyway

Reply:
> 
> 
> not true defibs no longer require ghost to be in body and you can return to it when your body comes back from death anyway

Since when?

Reply:
idk since like a few months but sure ill look into it more